state/factory/workingset.go

```diff
@@ -343,6 +343,9 @@ func (ws *workingSet) State(s interface{}, opts ...protocol.StateOption) (uint64
 	if err != nil {
 		return ws.height, err
 	}
+	if cfg.Keys != nil {
```

Comment:
why add this?

Reply:
it's in the removed `stateAtHeight()` func, it's a safety check, there's a similar check in `States()` at L